daemon/main.go

```diff
@@ -823,7 +823,13 @@ func runDaemon() {
 	}
 
 	if option.Config.RestoreState {
-
+		// Block on regenerating endpoints until we receive all policies from K8s
+		// if K8s is enabled.
+		if k8s.IsEnabled() {
```

Comment:
I probably didn't make this clear enough in the original issues. This needs to be moved to the main path and not be dependent on restore state. Even if we don't restore endpoints, new endpoints should not be created until we understand whether policy exists or not.

Reply:
OK, so we should also wait before starting regenerations in the regeneration queue. Got it.